pyboy/plugins/game_wrapper_pokemon_gen1.py

```diff
@@ -36,6 +71,22 @@ def post_tick(self):
         self._tile_cache_invalid = True
         self._sprite_cache_invalid = True
 
+        self.number_of_pokemon = self.pyboy.memory[ADDR_NUMBER_OF_POKEMON]
+        self.total_items = self.pyboy.memory[ADDR_TOTAL_ITEMS]
+        # Extract the time in game:
+        _hours = self.pyboy.memory[ADDR_HOURS]
+        _minutes = self.pyboy.memory[ADDR_MINUTES]
+        _seconds = self.pyboy.memory[ADDR_SECONDS]
+        self.game_time = f'{_hours}:{_minutes}:{_seconds}'
+
+        # Check whether the player is in battle mode or out of battle
+        if self.pyboy.memory[ADDR_BATTLE] == 1:
+            self.battle_state = 'In Battle'
+        else:
+            self.battle_state = 'Overworld'
```

Comment:
Did you test this? I think there's some distinction between trainer battle and wild pokémon.